client/lib/wpcom-xhr-wrapper/index.js

```diff
@@ -11,12 +9,14 @@ import debugModule from 'debug';
 const debug = debugModule( 'calypso:wpcom-xhr-wrapper' );
 
 export default function ( params, callback ) {
-	return xhr( params, function ( error, response, headers ) {
-		if ( error && error.name === 'InvalidTokenError' ) {
-			debug( 'Invalid token error detected, authorisation probably revoked - logging out' );
-			require( 'lib/user/utils' ).logout();
-		}
+	return import( /* webpackChunkName: "wpcom-xhr-request" */ 'wpcom-xhr-request' ).then( ( xhr ) =>
```

Comment:
This approach works well to allow for loading `wpcom-xhr-request` only when needed, great work! 👍 This function was already async (through callbacks), so this won't affect any consumers.

The ICFY results confirm that `wpcom-xhr-request` is not being added to the critical path.